python/ray/serve/_private/autoscaling_policy.py

```diff
@@ -46,6 +46,17 @@ def calculate_desired_num_replicas(
     smoothed_error_ratio = 1 + ((error_ratio - 1) * autoscaling_config.smoothing_factor)
     desired_num_replicas = math.ceil(current_num_replicas * smoothed_error_ratio)
 
+    # If error_ratio = 0 meaning there is no more traffic, and desired
+    # num replicas is stuck at a positive number of the math.ceil above,
+    # decrease desired_num_replicas by one so that the deployment can
+    # eventually scale to 0.
+    if (
+        error_ratio == 0
```

Comment:
is it possible we are stuck at downscaling when error_ratio is not equal to 0? (some value less than 1)

Reply:
Yeah that's possible, but I think that might just be expected behavior (not really getting "stuck"), since if error_ratio is not equal to 0 it means there is still traffic, so depending on the autoscaling parameters that are set, it's possible we just make the decision to not decrease the number of replicas.